### PR TITLE
Fix DBLPFetcherTest -- dblp changed the format

### PR DESCRIPTION
**Pull request by koppor**
dblp changed the format of URLs - `_` is now escaped.

Clean URLs in DBLPFetcher.

Additionally:

- Style RemoveLatexCommandsFormatter
- Add tests for RemoveLatexCommandsFormatter
- Add support for removing single and multiple whitespaces after a command
- Split tests in RemoveBracketsTest
- Fix casing in CleanupUrlFormatter (to match style and current name of test class CleanUpFormatterTest)
- Fix casing in some ...Url.. method names
- Format CleanupUrlFormatter

### CODE REVIEW

**Comment by Siedlerchr** [comment]
The fetcher should not store escaped URLs. The URL field is treated as verbatim field in biblatex, so no escaping necessarily

**Comment by koppor** [comment]
OK, I included the `CleanupUrlFormatter` and adapted it accordingly. While doing that, I also added some tests. See details in the updated PR description.

File: src/main/java/org/jabref/logic/formatter/bibtexfields/CleanupUrlFormatter.java [hunk 1]
```diff
@@ -38,16 +39,16 @@ public String format(String value) {
...
         try {
             decodedLink = URLDecoder.decode(toDecode, StandardCharsets.UTF_8.name());
-        }
-        catch (UnsupportedEncodingException e) {
+        } catch (UnsupportedEncodingException e) {
             LOGGER.warn("Used unsupported character encoding", e);
         }
-        return decodedLink;
+
+        String result = new RemoveLatexCommandsFormatter().format(decodedLink);
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/logic/formatter/bibtexfields/CleanupUrlFormatter.java]: Although somewhat of an edge case, but doesn't this lead to problems with the valid url `jabref.org/test%5Ctest` (which I guess would be reduced to `jabref.org/test` instead of the expected `jabref.org/test\test`?

Comment (koppor) [reply src/main/java/org/jabref/logic/formatter/bibtexfields/CleanupUrlFormatter.java]: Not this LOC in particularly, but in general: Yes. I added a test case for that.

In the fetcher itself, I use the RemoveLatexCommandsFormatter directly. I did not know how to convert a LayoutFormatter to a Formatter. Thus, I "manually" updated the field in cbb3836. Moments later, I worked on `LayoutFormatterBasedFormatter` which wraps a `LayoutFormatter` in a `Formatter` and IMHO made the code more nice.

Comment (tobiasdiez) [reply src/main/java/org/jabref/logic/formatter/bibtexfields/CleanupUrlFormatter.java]: I'm still unsure if we should really remove latex commands from the URL. I guess this is only an issue for the DBLP fetcher and nowhere else.

Comment (koppor) [reply src/main/java/org/jabref/logic/formatter/bibtexfields/CleanupUrlFormatter.java]: Oh, now I got it. Reverted that in https://github.com/JabRef/jabref/pull/5582/commits/83cbb3d02ce679ce6f6e5daf88d8510b3a804ac3
